cpp/src/strings/padding.cu

```diff
@@ -29,142 +26,135 @@
-  auto d_strings      = *strings_column;
-
-  // create null_mask
-  rmm::device_buffer null_mask = cudf::detail::copy_bitmask(strings.parent(), stream, mr);
-
-  // build offsets column
-  auto offsets_transformer_itr = cudf::detail::make_counting_transform_iterator(
-    0, compute_pad_output_length_fn{d_strings, width, fill_char_size});
-  auto offsets_column = make_offsets_child_column(
-    offsets_transformer_itr, offsets_transformer_itr + strings_count, stream, mr);
-  auto d_offsets = offsets_column->view().data<int32_t>();
-
-  // build chars column
-  auto const bytes =
-    cudf::detail::get_value<int32_t>(offsets_column->view(), strings_count, stream);
-  auto chars_column = strings::detail::create_chars_child_column(bytes, stream, mr);
-  auto d_chars      = chars_column->mutable_view().data<char>();
-
-  if (side == side_type::LEFT) {
-    thrust::for_each_n(
-      rmm::exec_policy(stream),
-      thrust::make_counting_iterator<cudf::size_type>(0),
-      strings_count,
-      [d_strings, width, d_fill_char, d_offsets, d_chars] __device__(size_type idx) {
-        if (d_strings.is_valid(idx)) {
-          pad_impl<side_type::LEFT>(
-            d_strings.element<string_view>(idx), width, d_fill_char, d_chars + d_offsets[idx]);
-        }
-      });
-  } else if (side == side_type::RIGHT) {
-    thrust::for_each_n(
-      rmm::exec_policy(stream),
-      thrust::make_counting_iterator<cudf::size_type>(0),
-      strings_count,
-      [d_strings, width, d_fill_char, d_offsets, d_chars] __device__(size_type idx) {
-        if (d_strings.is_valid(idx)) {
-          pad_impl<side_type::RIGHT>(
-            d_strings.element<string_view>(idx), width, d_fill_char, d_chars + d_offsets[idx]);
-        }
-      });
-  } else if (side == side_type::BOTH) {
-    thrust::for_each_n(
-      rmm::exec_policy(stream),
-      thrust::make_counting_iterator<cudf::size_type>(0),
-      strings_count,
-      [d_strings, width, d_fill_char, d_offsets, d_chars] __device__(size_type idx) {
-        if (d_strings.is_valid(idx)) {
-          pad_impl<side_type::BOTH>(
-            d_strings.element<string_view>(idx), width, d_fill_char, d_chars + d_offsets[idx]);
-        }
-      });
-  }
+  auto d_fill_char          = char_utf8{0};
+  auto const fill_char_size = to_char_utf8(fill_char.data(), d_fill_char);
+
+  auto d_strings = column_device_view::create(input.parent(), stream);
+
+  auto children = [&] {
+    if (side == side_type::LEFT) {
+      auto fn = pad_fn<side_type::LEFT>{*d_strings, width, fill_char_size, d_fill_char};
+      return make_strings_children(fn, input.size(), stream, mr);
+    } else if (side == side_type::RIGHT) {
+      auto fn = pad_fn<side_type::RIGHT>{*d_strings, width, fill_char_size, d_fill_char};
+      return make_strings_children(fn, input.size(), stream, mr);
+    }
+    auto fn = pad_fn<side_type::BOTH>{*d_strings, width, fill_char_size, d_fill_char};
+    return make_strings_children(fn, input.size(), stream, mr);
```

Comment:
Nit: Make `auto const fn = ...`.

Reply:
This cannot be made `const` since it is modified by `make_strings_children`

Reply:
Isn't the functor `fn` passed by copy to `make_strings_children()`? Perhaps I've misunderstood why this can't be const.

Reply:
Yeah it should be passed by copying.

Reply:
I just tested that. There is no moving: https://godbolt.org/z/s73nvcjnq.

Reply:
I think making this `const` incorrectly communicates its usage. Is changing this required for approving the PR?

Reply:
Nah just a nit. Coding style BTW.